File: apps/desktop/src/desktop/talk.ts
import { invoke } from "@tauri-apps/api/core";
import { isTauriRuntime } from "./tauri.ts";

export type DesktopTalkFixtureAudio = {
  bytes: Uint8Array;
  fileName: string;
  mimeType: string;
};

type DesktopTalkFixturePayload = {
  bytes: number[];
  fileName: string;
  mimeType: string;
};

export async function loadDesktopTalkFixtureAudio(): Promise<DesktopTalkFixtureAudio | null> {
  if (!isTauriRuntime()) {
    return null;
  }
  const result = await invoke<DesktopTalkFixturePayload>("desktop_talk_test_fixture_audio");
  return {
    bytes: Uint8Array.from(result.bytes),
    fileName: result.fileName,
    mimeType: result.mimeType,
  };
}

File: apps/desktop/src-tauri/src/talk.rs
use serde::Serialize;

#[derive(Serialize)]
#[serde(rename_all = "camelCase")]
pub struct DesktopTalkFixtureAudio {
  pub bytes: Vec<u8>,
  pub file_name: String,
  pub mime_type: String,
}

#[cfg(debug_assertions)]
const SKY_VOICE_TEST_WAV: &[u8] = include_bytes!("../../../../test-fixtures/audio/sky_voice_test.wav");

#[tauri::command]
pub fn desktop_talk_test_fixture_audio() -> Result<DesktopTalkFixtureAudio, String> {
  #[cfg(debug_assertions)]
  {
    return Ok(DesktopTalkFixtureAudio {
      bytes: SKY_VOICE_TEST_WAV.to_vec(),
      file_name: "sky_voice_test.wav".to_string(),
      mime_type: "audio/wav".to_string(),
    });
  }

  #[cfg(not(debug_assertions))]
  {
    Err("desktop talk fixture audio is only available in debug builds".to_string())
  }
}
